File: src/java/io/Reader.ts
export abstract class Reader {
  // Constructors
  public abstract constructor() {
    return new Packages.java.io.Reader()
  }
  // Constructors
  public abstract constructor(obj: Object) {
    return new Packages.java.io.Reader(obj)
  }
  // Methods
  public abstract close(): void {
    this.close()
  }
  public mark(int: number): void {
    this.mark(int)
  }
  public markSupported(): boolean {
    return this.markSupported()
  }
  public read(): number {
    return this.read()
  }
  public read(cha: CharBuffer): number {
    return this.read(cha)
  }
  public read(c: String[]): number {
    return this.read(c)
  }
  public abstract read(c: String[], int: number, int_1: number): number {
    return this.read(c, int, int_1)
  }
  public ready(): boolean {
    return this.ready()
  }
  public reset(): void {
    this.reset()
  }
  public skip(lon: number): number {
    return this.skip(lon)
  }
  // Fields
  protected lock: Object
  private skipBuffer: String[]
  private static maxSkipBufferSize: number
}

File: src/java/io/Reader.java
public abstract class java.io.Reader
{
    // Constructors
    public abstract java.io.Reader();
    public abstract java.io.Reader(java.lang.Object);
    // Methods
    public abstract void close();
    public void mark(int);
    public boolean markSupported();
    public int read();
    public int read(java.nio.CharBuffer);
    public int read([C);
    public abstract int read([C, int, int);
    public boolean ready();
    public void reset();
    public long skip(long);
    // Fields
    protected java.lang.Object lock;
    private [C skipBuffer;
    private static final int maxSkipBufferSize;
}